awx/main/validators.py

```diff
@@ -195,3 +195,37 @@ def vars_validate_or_raise(vars_str):
         return vars_str
     except ParseError as e:
         raise RestValidationError(str(e))
+
+
+def validate_container_image_name(value):
+    """
+    from https://github.com/distribution/distribution/blob/af8ac809336c2316c81b08605d92d94f8670ad15/reference/reference.go#L4
+
+    Grammar
+
+    reference                       := name [ ":" tag ] [ "@" digest ]
+    name                            := [domain '/'] path-component ['/' path-component]*
+    domain                          := domain-component ['.' domain-component]* [':' port-number]
+    domain-component                := /([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])/
+    port-number                     := /[0-9]+/
+    path-component                  := alpha-numeric [separator alpha-numeric]*
+    alpha-numeric                   := /[a-z0-9]+/
+    separator                       := /[_.]|__|[-]*/
+
+    tag                             := /[\w][\w.-]{0,127}/
+
+    digest                          := digest-algorithm ":" digest-hex
+    digest-algorithm                := digest-algorithm-component [ digest-algorithm-separator digest-algorithm-component ]*
+    digest-algorithm-separator      := /[+.-_]/
+    digest-algorithm-component      := /[A-Za-z][A-Za-z0-9]*/
+    digest-hex                      := /[0-9a-fA-F]{32,}/ ; At least 128 bit digest value
+
+    identifier                      := /[a-f0-9]{64}/
+    short-identifier                := /[a-f0-9]{6,64}/
+
+    The regex below is the printed value of the following GO variable, which represents the 'reference' line above, i.e. the full image name + tag or digest
+    https://github.com/distribution/distribution/blob/af8ac809336c2316c81b08605d92d94f8670ad15/reference/regexp.go#L72
+    """
+    regex = r"^((?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?\/)?[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:\/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)(?::([\w][\w.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][0-9a-fA-F]{32,}))?$"
```

Comment:
one big ugly regex, but I put some comments above on how I obtained this.

Reply:
![mother_of_god](https://user-images.githubusercontent.com/28930622/122057529-bcca8e80-cdb8-11eb-96a3-3f266928fae8.gif)

Reply:
I started to give this advice in Slack, but I think that this regex should be turned into an indented full-block regex, for readability.

Reply:
https://docs.python.org/3/howto/regex.html#using-re-verbose
https://pymotw.com/3/re/#verbose-expression-syntax